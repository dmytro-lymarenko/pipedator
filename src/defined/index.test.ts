import { defined } from './index';

describe('defined()', () => {
	it('should validate a defined value', () => {
		expect(defined().validate('')).toBeNull();
		expect(defined().validate(0)).toBeNull();
		expect(defined().validate([])).toBeNull();
		expect(defined().validate({})).toBeNull();
	});

	it('should fail when value either undefined or null', () => {
		expect(defined().validate(null)).toMatchSnapshot();
		expect(defined().validate(undefined)).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(defined('Custom message').validate(undefined)).toMatchSnapshot();
	});
});
