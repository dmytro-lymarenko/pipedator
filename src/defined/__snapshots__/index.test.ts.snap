// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defined() should fail when value either undefined or null 1`] = `
Object {
  "children": null,
  "message": "Value should be defined",
  "path": "",
}
`;

exports[`defined() should fail when value either undefined or null 2`] = `
Object {
  "children": null,
  "message": "Value should be defined",
  "path": "",
}
`;

exports[`defined() should support custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
